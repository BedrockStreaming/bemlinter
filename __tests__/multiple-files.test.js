const { lint, format } = require('../src/bemlinter.js');

const snapLintOutput = (fileName, done, options = {}) => {
  lint(`${__dirname}/styles/${fileName}`, options)
    .then(lintResult => format(lintResult, false))
    .then((output) => {
      expect(output).toMatchSnapshot();
    })
    .catch(console.error)
    .then(done)
  ;
};

describe('Bemlinter of crossed styled files', () => {
  it('should log error on both blocks', done => snapLintOutput('cross-styling/*.scss', done, { classPrefix: 'c-' }));

  it('should not log error on the external block', done => snapLintOutput('cross-styling/*.scss', done, {
    excludeBlock: ['external'],
    classPrefix: 'c-',
  }));
});

describe('Bemlinter of multi-modules files', () => {
  it('should detect the module and the missing prefix', done => snapLintOutput('mixed-settings/*.scss', done, {
    modules: [{
      name: 'module',
      sources: ['./__tests__/styles/mixed-settings/module-prefixed.scss'],
      filePattern: 'module-([^.]*)\.scss',
    }],
  }));

  it('should detect the module and the associate leak styles', done => snapLintOutput('mixed-settings/*.scss', done, {
    modules: [{
      name: 'module',
      sources: ['./__tests__/styles/mixed-settings/module-prefixed.scss'],
      classPrefix: 'c-',
      filePattern: 'module-([^.]*)\.scss',
    }],
  }));
});
