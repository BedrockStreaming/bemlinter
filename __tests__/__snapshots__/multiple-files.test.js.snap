exports[`Bemlinter of crossed styled files should log error on both blocks 1`] = `
"  ✗ external
[leak.scss:8] Error: \".c-external\" should not be styled outside of its own stylesheet.
  ✗ leak
[leak.scss:8] Error: \".c-external\" is incoherent with the file name, \".c-leak\" expected.

FAIL: bemlinter has detected 2 errors on 2 blocks."
`;

exports[`Bemlinter of crossed styled files should not log error on the external block 1`] = `
"  ✗ leak
[leak.scss:8] Error: \".c-external\" is incoherent with the file name, \".c-leak\" expected.

FAIL: bemlinter has detected 1 error on 1 block."
`;

exports[`Bemlinter of multi-modules files should detect the module and the associate leak styles 1`] = `
"  ✗ unprefixed
[unprefixed.scss:8] Error: \".c-prefixed\" is incoherent with the file name, \".unprefixed\" expected.
  ✗ module / prefixed
[unprefixed.scss:8] Error: \".c-prefixed\" should not be styled outside of its own stylesheet.

FAIL: bemlinter has detected 2 errors on 2 blocks."
`;

exports[`Bemlinter of multi-modules files should detect the module and the missing prefix 1`] = `
"  ✗ unprefixed
[unprefixed.scss:8] Error: \".c-prefixed\" is incoherent with the file name, \".unprefixed\" expected.
  ✗ module / prefixed
[module-prefixed.scss:1] Error: \".c-prefixed\" is incoherent with the file name, \".prefixed\" expected.

FAIL: bemlinter has detected 2 errors on 2 blocks."
`;
